File: app/src/main/java/com/personal/accountantAssistant/utils/ParserUtils.java
package com.personal.accountantAssistant.utils;

import com.personal.accountantAssistant.ui.bills.entities.Bills;
import com.personal.accountantAssistant.ui.buys.entities.Buys;
import com.personal.accountantAssistant.ui.payments.entities.Payments;

import java.util.List;

public class ParserUtils {

    public static boolean isNullObject(final Object object) {
        return (object == null);
    }

    public static boolean isNotNullObject(final Object object) {
        return !isNullObject(object);
    }

    public static boolean isNotNullAndNotEmpty(final List<?> list) {
        return list != null &&
                !list.isEmpty();
    }

    public static boolean isNotNullAndNotEmpty(final String stringValue) {
        return stringValue != null &&
                !stringValue.isEmpty();
    }

    private static boolean isNullOrEmpty(final List<?> list) {
        return list == null ||
                list.isEmpty();
    }

    private static boolean isBuyInstance(final Object entity) {
        return entity instanceof Buys;
    }

    private static boolean isBillInstance(final Object entity) {
        return entity instanceof Bills;
    }

    public static Bills toBills(final Object entity) {
        if (isBillInstance(entity)) {
            return (Bills) entity;
        } else {
            return null;
        }
    }

    public static Buys toBuys(final Object entity) {
        if (isBuyInstance(entity)) {
            return (Buys) entity;
        } else {
            return null;
        }
    }

    public static Payments toPayments(final Object entity) {
        Payments payments = null;
        if (isNotNullObject(entity)) {
            if (ParserUtils.isBuyInstance(entity)) {
                payments = new Payments((Buys) entity);
            } else if (ParserUtils.isBillInstance(entity)) {
                payments = new Payments((Bills) entity);
            } else if (ParserUtils.isPaymentInstance(entity)) {
                payments = (Payments) entity;
            }
        }
        return payments;
    }

    private static boolean isPaymentInstance(Object entity) {
        return entity instanceof Payments;
    }
}

File: app/src/main/java/com/personal/accountantAssistant/utils/ParserUtils.kt
package com.personal.accountantAssistant.utils

import com.personal.accountantAssistant.ui.bills.entities.Bills
import com.personal.accountantAssistant.ui.buys.entities.Buys
import com.personal.accountantAssistant.ui.payments.entities.PaymentsEntity
import com.personal.accountantAssistant.ui.wallet.entities.CardEntity

object ParserUtils {
    @JvmStatic
    fun isNullObject(`object`: Any?): Boolean {
        return `object` == null
    }

    private fun isCardInstance(entity: Any?): Boolean {
        return entity is CardEntity
    }

    private fun isBuyInstance(entity: Any?): Boolean {
        return entity is Buys
    }

    private fun isBillInstance(entity: Any?): Boolean {
        return entity is Bills
    }

    fun toPayments(entity: Any?) = when {
        isBuyInstance(entity) -> PaymentsEntity(entity as Buys)
        isBillInstance(entity) -> PaymentsEntity(entity as Bills)
        isPaymentInstance(entity) -> entity as PaymentsEntity
        else -> null
    }

    private fun isPaymentInstance(entity: Any?): Boolean {
        return entity is PaymentsEntity
    }
}